pkg/xen-tools/patches-4.15.0/13-qemu-Add-cpu-pin-and-cpumask-options.patch

```diff
@@ -0,0 +1,128 @@
+
+Signed-off-by: Nikolay Martyanov <ohmspectator@gmail.com>
+---
+ tools/qemu-xen/hw/core/machine.c   | 38 ++++++++++++++++++++++++++++++
+ tools/qemu-xen/include/hw/boards.h |  2 ++
+ tools/qemu-xen/qemu-options.hx     |  8 +++++++
+ tools/qemu-xen/softmmu/vl.c        |  4 ++++
+ 4 files changed, 52 insertions(+)
+
+diff --git a/tools/qemu-xen/hw/core/machine.c b/tools/qemu-xen/hw/core/machine.c
+index 8d1a90c..2f9993e 100644
+--- a/tools/qemu-xen/hw/core/machine.c
++++ b/tools/qemu-xen/hw/core/machine.c
+@@ -372,6 +372,35 @@ static void machine_set_graphics(Object *obj, bool value, Error **errp)
+     ms->enable_graphics = value;
+ }
+ 
++static bool machine_get_cpu_pin(Object *obj, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    return ms->cpu_pin;
++}
++
++static void machine_set_cpu_pin(Object *obj, bool value, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    ms->cpu_pin = value;
++}
++
++static char *machine_get_cpumask(Object *obj, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    return g_strdup(ms->cpumask_str);
++}
++
++static void machine_set_cpumask(Object *obj, const char *value, Error **errp)
++{
++    MachineState *ms = MACHINE(obj);
++
++    g_free(ms->cpumask_str);
++    ms->cpumask_str = g_strdup(value);
++}
++
+ static char *machine_get_firmware(Object *obj, Error **errp)
+ {
+     MachineState *ms = MACHINE(obj);
+@@ -841,6 +870,15 @@ static void machine_class_init(ObjectClass *oc, void *data)
+     object_class_property_set_description(oc, "usb",
+         "Set on/off to enable/disable usb");
+ 
++    object_class_property_add_bool(oc, "cpu-pin",
++        machine_get_cpu_pin, machine_set_cpu_pin);
++    object_class_property_set_description(oc, "cpu-pin",
++        "Set on/off to enable/disable CPU pinning");
++
++    object_class_property_add_str(oc, "cpumask",
++        machine_get_cpumask, machine_set_cpumask);
++    object_class_property_set_description(oc, "cpumask", "CPU Mask");
++
+     object_class_property_add_bool(oc, "graphics",
+         machine_get_graphics, machine_set_graphics);
+     object_class_property_set_description(oc, "graphics",
+diff --git a/tools/qemu-xen/include/hw/boards.h b/tools/qemu-xen/include/hw/boards.h
+index 426ce5f..b06f13e 100644
+--- a/tools/qemu-xen/include/hw/boards.h
++++ b/tools/qemu-xen/include/hw/boards.h
+@@ -294,6 +294,8 @@ struct MachineState {
+     char *kernel_filename;
+     char *kernel_cmdline;
+     char *initrd_filename;
++    bool cpu_pin;
++    char *cpumask_str;
+     const char *cpu_type;
+     AccelState *accelerator;
+     CPUArchIdList *possible_cpus;
+diff --git a/tools/qemu-xen/qemu-options.hx b/tools/qemu-xen/qemu-options.hx
+index 708583b..d610528 100644
+--- a/tools/qemu-xen/qemu-options.hx
++++ b/tools/qemu-xen/qemu-options.hx
+@@ -509,6 +509,14 @@ SRST
+     Preallocate memory when using -mem-path.
+ ERST
+ 
++DEF("cpu-pin", 0, QEMU_OPTION_cpu_pin,
++    "-cpu-cpu   pin any VPCU thread to a PCPU\n",
```

Comment:
PCPU is not mentioned in any qemu command and I've seen people using pCPU and PCPU. Although I believe should be understandable for the majority of users, what do you think to use "Physical CPU" or "PCPU (Physical CPU)"? wouldn't be better for understanding?

VCPU is already mentioned in commands (with capital V), so I think it's fine to keep it.

Reply:
I'll change it to "physical CPU" and fix a typo. Thanks!